### PR TITLE
Set CSGPolygon path_interval step size and minimum to 0.01

### LINKED ISSUE #52179: New CSG Polygon bugs introduced by "Fix multiple issues with CSGPolygon #49314"
### Godot version

3.4 beta 4

### System information

Windows 10, GLES3

### Issue description

Regarding https://github.com/godotengine/godot/pull/49314 I have run into a few issues.

1) Interval is now a fraction of the entire length of the curve (???)* rather than a distance, meaning long curves behave different from short curves with the same settings as far as polygon density goes.  This worked better as a distance in meters and was consistent with the curve bake interval. *Not sure if it's actually a fraction, because .1 seems like it would be broken into 10ths, but there are more than 10 segments, and sometimes a value of 1 has more than 1 segment.

2) Can't get the UV's to behave the same as before.  The UV behavior is probably better... mostly, but still has oddities.
Before:
image
After:
image
On the plus side, it tries to wrap the texture around.  On the minus side, it seems to only use 1/2 of the texture:
image
I would expect this to wrap around entirely so I could have a tiled pipe texture or something.  Maybe it's because the other half of the texture is reserved for the caps?  Wonder if that should be an option.
Also, the texture coverage doesn't take the size of the polygon into account, so a really small face uses the same amount of texture as a really large one:
image
image
Also, it looks like the UV's reset to 0 for each segment, which works OK for static textures, but not for custom shaders (ex: a pulse that flows across a cable).  And, since the segment length is a fraction of the total curve, the overall texture coverage, and thus timing of pulses and other shader effects, don't cover a set distance using the UV's like it used to.

3) Limit of .1 for the interval is insufficient for long curves.
image
Pretty self-explanatory.  I need more detail for my long cables!  I actually had an engine modification to simplify based on the curve angle which I need to make a pull request for (after I update it to match the new code).

### Steps to reproduce

Make a really long path and try to make the CSG geometry follow it and look good. :)

### Minimal reproduction project

[test_csg_path_3.4b4.zip](https://github.com/godotengine/godot/files/7070142/test_csg_path_3.4b4.zip)
[test_csg_path_3.3.3.zip](https://github.com/godotengine/godot/files/7070143/test_csg_path_3.3.3.zip)

### PR DESCRIPTION
**Pull request by madmiraal**
~Also fixes `MODE_PATH` extrusion count adjustment for joined paths.~

Fixes #52179.

**Note:** #52179 includes a use case for requiring the texture to repeat by a specified distance, which is not addressed by this PR.

### CODE REVIEW

File: modules/csg/csg_shape.cpp [hunk 1]
```diff
@@ -1745,9 +1745,10 @@ CSGBrush *CSGPolygon3D::_build_brush() {
 			break;
 		case MODE_PATH: {
 			extrusions = Math::ceil(1.0 * curve->get_point_count() / path_interval);
-			if (!path_joined) {
+			if (path_joined) {
+				extrusions += 1;
+			} else {
 				end_count = 2;
-				extrusions -= 1;
```
Comment (jitspoe) [inline modules/csg/csg_shape.cpp:1750]: Why did you change this part?  If the path_interval is 1, you should just have 1 segment, correct?  But there are 2 curve points, so you'll get a subdivision even if the interval is set to 1.

(side note: I'm not sure if using the number of points on the curve makes sense at all, as the subdivisions won't match up 1:1 with the curve points).

Comment (madmiraal) [reply modules/csg/csg_shape.cpp]: I thought it was undercounting, but it isn't. I'll change it back.

**Comment by akien-mga** [comment]
> Fixes #52179.

I don't think this PR fixes #52179, given the discussion there and the much more substantial PRs that @jitspoe has opened to address shortcomings after #49314.

**Comment by akien-mga** [comment]
Superseded by #52512.
